### ISSUE TITLE
[Bug] A11y: SelectPanel listbox does not have an aria label

### ISSUE THREAD

**Issue by mijpeterson** [primer/react#2215]
**Describe the bug**
When using the SelectPanel component, it appears that we are unable to configure the listbox it renders with an aria label.  This means that it does not have an accessible name.

Digging into the code, I've found that the SelectPanel does spread the rest of its props over to the FilteredActionList component, which then gets passed down to the ActionList component.  So if we were able to supply an "aria-label" or "title" prop to SelectPanel, then we'd be able to solve this bug.  However, the SelectPanel's type definition for its props does not accept List props.

**To Reproduce**
1. Go to [SelectPanel docs](https://primer.style/react/SelectPanel)
2. Expand one of the example SelectPanels 
3. Open developer tools

**Expected behavior**
"title" or "aria-label" is defined for the listbox.

**Screenshots**
![A side by side view of the expanded SelectPanel and its HTML elements in the browser dev tools. The dev tools highlight that the element is missing an accessible name.](https://user-images.githubusercontent.com/110266616/182893262-5ba69741-f528-456f-962a-48df4b2df850.png)

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Edge
 - Version: 103.0.1264.77

**Additional context**
The easier way to solve this issue would be to expose one of "aria-label" or "title" through the SelectPanel's props type.  However, given that the ActionList component that it uses under the hood is deprecated, this might be a good chance to switch over the internals.

**Comment by siddharthkp** [comment]
Hi! Thanks for taking the time to open an issue, I have added it to our backlog to look into!

**Comment by mijpeterson** [comment]
Commenting to keep the issue open.

**Comment by mijpeterson** [comment]
Commenting to keep the issue open.

**Comment by lesliecdubs** [comment]
@siddharthkp do you think any of our active work right now on `SelectPanel` might resolve this?

**Comment by lesliecdubs** [comment]
Great! Let's leave it open until then. I added it to the Primer Component Remediation project board under `SelectPanel` so we'll remember to review/close this once the new component is ready for prime time.

**Comment by mijpeterson** [comment]
Once again commenting to keep this open.
